src/operations/operation.ts

```diff
@@ -69,6 +73,11 @@ export abstract class AbstractOperation<TResult = any> {
     this.bsonOptions = resolveBSONOptions(options);
 
     this[kSession] = options.session != null ? options.session : undefined;
+    if (options?.session?.explicit && options?.session?.timeoutMS != null && options.timeoutMS) {
```

Comment:
Just a thought - it might make more sense to put this in `execute_operation`, where we already have other logic that throws with invalid session configurations:

```typescript
  if (session == null) {
    owner = Symbol();
    session = client.startSession({ owner, explicit: false });
  } else if (session.hasEnded) {
    throw new MongoExpiredSessionError('Use of expired sessions is not permitted');
  } else if (session.snapshotEnabled && !topology.capabilities.supportsSnapshotReads) {
    throw new MongoCompatibilityError('Snapshot reads require MongoDB 5.0 or later');
  } else if (session.client !== client) {
    throw new MongoInvalidArgumentError('ClientSession must be from the same MongoClient');
  }
```

Reply:
Good callout here.